### PR TITLE
ci: keep opi-api version in built image's label

### PR DESCRIPTION
**Pull request by yanjing1104**
fix https://github.com/opiproject/opi-spdk-bridge/issues/576

### CODE REVIEW

**Comment by yanjing1104** [comment]
This PR is used to replace https://github.com/opiproject/opi-spdk-bridge/pull/590 (which can not be re-opened cause I deleted the forked repository in the middle), the docker image is tested in https://github.com/yanjing1104/opi-spdk-bridge/tree/test1.

```bash
# docker pull ghcr.io/yanjing1104/opi-spdk-bridge:pr-1
# docker image ls 
REPOSITORY                                         TAG             IMAGE ID       CREATED          SIZE
ghcr.io/yanjing1104/opi-spdk-bridge                pr-1            dbf394cef7a6   11 minutes ago   52.4MB
#  docker inspect dbf394cef7a6 --format '{{ json .Config.Labels }}' | sed 's/,/\n/g'
{"opi-api-version":"v0.0.0-20230826011814-48b273e85a4f"
"org.opencontainers.image.created":"2023-08-29T07:31:50.090Z"
"org.opencontainers.image.description":"OPI Storage gRPC to SPDK json-rpc bridge"
"org.opencontainers.image.licenses":"Apache-2.0"
"org.opencontainers.image.revision":"02c47eceabb37d3cf367559bb49f85712777b5c0"
"org.opencontainers.image.source":"https://github.com/yanjing1104/opi-spdk-bridge"
"org.opencontainers.image.title":"opi-spdk-bridge"
"org.opencontainers.image.url":"https://github.com/yanjing1104/opi-spdk-bridge"
"org.opencontainers.image.version":"pr-1"}
```

File: Dockerfile [hunk 1]
```diff
@@ -3,6 +3,9 @@
 
 FROM docker.io/library/golang:1.21.0-alpine as builder
 
+ARG OPI_API_VERSION
+LABEL opi-api-version=${OPI_API_VERSION}
```
Comment (artek-koltun) [inline Dockerfile:7]: Do we still need this part, since we do not set the version as a build argument in CI?

**Review by artek-koltun — COMMENTED** [review]
Thank you, Jing, for the patch!

File: Dockerfile (hunk 1 above)
Comment (yanjing1104) [reply Dockerfile]: No we do not need it, updated this PR with deleting the ARG and LABEL in Dockerfile.

Tested in https://github.com/yanjing1104/opi-spdk-bridge/tree/test2 

```bash
# docker pull ghcr.io/yanjing1104/opi-spdk-bridge:pr-2
pr-2: Pulling from yanjing1104/opi-spdk-bridge
7264a8db6415: Already exists 
6b2082003f5c: Pull complete 
e9d6d61eb267: Pull complete 
Digest: sha256:4f4c8e36261efe0f9f2a14d7c16edf191234c96273b15077af3afefb09439d74
Status: Downloaded newer image for ghcr.io/yanjing1104/opi-spdk-bridge:pr-2
ghcr.io/yanjing1104/opi-spdk-bridge:pr-2

# docker image ls | head -n2
REPOSITORY                                         TAG             IMAGE ID       CREATED             SIZE
ghcr.io/yanjing1104/opi-spdk-bridge                pr-2            b4eb501e442f   12 minutes ago      52.4MB

# docker inspect b4eb501e442f --format '{{ json .Config.Labels }}' | sed 's/,/\n/g'
{"opi-api-version":"v0.0.0-20230826011814-48b273e85a4f"
"org.opencontainers.image.created":"2023-08-29T08:33:19.060Z"
"org.opencontainers.image.description":"OPI Storage gRPC to SPDK json-rpc bridge"
"org.opencontainers.image.licenses":"Apache-2.0
"
"org.opencontainers.image.revision":"d8d96d366dafb03bd0d0cf72d46b4811de328abc"
"org.opencontainers.image.source":"https://github.com/yanjing1104/opi-spdk-bridge"
"org.opencontainers.image.title":"opi-spdk-bridge"
"org.opencontainers.image.url":"https://github.com/yanjing1104/opi-spdk-bridge"
"org.opencontainers.image.version":"pr-2"}
```

**Review by artek-koltun — APPROVED** [review]
LGTM

**Review by glimchb — APPROVED** [review]
Lgtm
@artek-koltun ?

**Pull request merged**